src/main/java/io/codeka/gaia/repository/TerraformModuleGitRepository.java

```diff
@@ -0,0 +1,64 @@
+package io.codeka.gaia.repository;
+
+import io.codeka.gaia.bo.TerraformModule;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Repository
+public class TerraformModuleGitRepository {
+
+    private static final Pattern GITHUB_PATTERN = Pattern.compile("^http[s]?://[www.]?github.com(.*).git$");
+    private static final Pattern GITLAB_PATTERN = Pattern.compile("^(http[s]?://[www.]?gitlab.*).git$");
+
+    private RestTemplate restTemplate;
+
+    @Autowired
+    public TerraformModuleGitRepository(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public Optional<String> getReadme(TerraformModule module) {
+        String url = StringUtils.EMPTY;
+        var matcher = GITHUB_PATTERN.matcher(module.getGitRepositoryUrl());
+        if (matcher.matches()) {
+            url = getGithubRawUrl(matcher, module);
+        }
+        matcher = GITLAB_PATTERN.matcher(module.getGitRepositoryUrl());
+        if (matcher.matches()) {
+            url = getGitlabRawUrl(matcher, module);
+        }
+
+        var response = restTemplate.exchange(
+                url + "/README.md",
+                HttpMethod.GET,
+                new HttpEntity<>(new HttpHeaders()),
+                String.class);
+        if (response.getStatusCode() == HttpStatus.OK) {
+            return Optional.ofNullable(response.getBody());
+        }
+        return Optional.empty();
+    }
+
+    private String getGithubRawUrl(Matcher matcher, TerraformModule module) {
+        return "https://raw.githubusercontent.com" + matcher.group(1) +
+                "/" + module.getGitBranch() +
+                "/" + module.getDirectory();
```

Comment:
I also have an issue when module's branch and moduleDirectory are empty.
The computed url is https://raw.githubusercontent.com/juwit/terraform-docker-mongo/null/null/README.md
Mayby branch should default to master, and we should null-check for the directory?